### PR TITLE
test-case: add support for testing on specified pipeline

### PR DESCRIPTION
**Pull request by aiChaoSONG**
with this patch, QA can run test case on specified single pipeline rather than iterate all pipelines

### CODE REVIEW

**Comment by Bin-QA** [comment]
@aiChaoSONG can you post the test result?
how to deal like: 'id:1,2,3'

**Comment by aiChaoSONG** [comment]
> @aiChaoSONG can you post the test result?
> how to deal like: 'id:1,2,3'

@Bin-QA I know you will ask me this, currently, only one pipeline is supported, if multiple id should be supported, we need to expose the filter string to user, and let user specified filter string

**Comment by Bin-QA** [comment]
> @Bin-QA I know you will ask me this, currently, only one pipeline is supported, if multiple id should be supported, we need to expose the filter string to user, and let user specified filter string

@aiChaoSONG [simultaneous-playback-capture.sh](https://github.com/thesofproject/sof-test/blob/master/test-case/simultaneous-playback-capture.sh#L58) can use like `id:1,2,3`
I think you need post the test result for it  `script -P "1,2,3"`

**Comment by libinyang** [comment]
pipeline ID is not the same of PCM ID. For user, it's hard and complicated to match the PCM id with pipeline id. I suppose @ClarexZhou would like use PCM id, right?

**Comment by Bin-QA** [comment]
> pipeline ID is not the same of PCM ID. For user, it's hard and complicated to match the PCM id with pipeline id. I suppose @ClarexZhou would like use PCM id, right?

the fact they are same value, you can check with [verify-pcm-list.sh](https://github.com/thesofproject/sof-test/blob/master/test-case/verify-pcm-list.sh)

File: case-lib/pipeline.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ func_pipeline_export()
         opt="-f"
         while [ $# -ne 0 ]
         do
-            opt=$opt" $1"
+            opt=$opt" '$1'"
```
Comment (marc-hb) [inline case-lib/pipeline.sh:23]: I don't know this code yet, could you give an example showing why this is required?

**Comment by libinyang** [comment]
> > pipeline ID is not the same of PCM ID. For user, it's hard and complicated to match the PCM id with pipeline id. I suppose @ClarexZhou would like use PCM id, right?
> 
> the fact they are same value, you can check with [verify-pcm-list.sh](https://github.com/thesofproject/sof-test/blob/master/test-case/verify-pcm-list.sh)

OK. Then here the pipeline id is actually the pcm id, not the real "pipeline id" in tplg.

File: case-lib/pipeline.sh (hunk 1 above)
Comment (aiChaoSONG) [reply case-lib/pipeline.sh:23]: [cmd_line_log.txt](https://github.com/thesofproject/sof-test/files/4479510/cmd_line_log.txt)
@marc-hb Take a look at the log attached, You should see why I add a `' '`

**Comment by Bin-QA** [comment]
> Is there a plan / github issue to move the `OPT_*` triplication to some common file like `lib.sh` or similar?

`OPT_*` at https://github.com/thesofproject/sof-test/blob/master/case-lib/opt.sh  
`opt.sh` will source at https://github.com/thesofproject/sof-test/blob/master/case-lib/lib.sh#L5

**Pull request merged**